packages/iris-grid/src/IrisGrid.jsx

```diff
@@ -1661,10 +1661,27 @@ export class IrisGrid extends Component {
     }
   }
 
+  isTableSearchAvailable() {
+    const { model, canToggleSearch } = this.props;
+    const searchDisplayMode = model?.layoutHints?.searchDisplayMode;
+
+    if (searchDisplayMode) {
+      if (searchDisplayMode === dh.SearchDisplayMode.SEARCH_DISPLAY_HIDE) {
+        return false;
+      }
+      if (searchDisplayMode === dh.SearchDisplayMode.SEARCH_DISPLAY_SHOW) {
+        return true;
+      }
+    }
```

Comment:
The outside if is unnecessary, could just be:
```suggestion
    if (searchDisplayMode === dh.SearchDisplayMode.SEARCH_DISPLAY_HIDE) {
      return false;
    }
    if (searchDisplayMode === dh.SearchDisplayMode.SEARCH_DISPLAY_SHOW) {
      return true;
    }
```